### ISSUE TITLE
Increasing file size does not work with daemon + Clamscan "INSTREAM: Size limit reached, (requested: 65536, max: 0)"

### ISSUE THREAD

**Issue by philly-vanilly** [kylefarris/clamscan#131]
I am using google-cloud-sdk/slim docker image with this config (note the non default size limits):

```
RUN apt-get install clamav-daemon -y && \
    npm ci && \
    curl -sSL https://sdk.cloud.google.com/ | bash && \
    echo "StreamMaxLength 40M" >> /etc/clamav/clamd.conf && \
    echo "MaxFileSize 40M" >> /etc/clamav/clamd.conf && \
    echo "MaxScanSize 40M" >> /etc/clamav/clamd.conf && \
    mkdir /unscanned_files
```

and

```
  try {
    if (clamscan === null) {
      clamscan = await (new NodeClam().init({
        removeInfected: false,
        quarantineInfected: false,
        scanLog: null,
        debugMode: true,
        fileList: null,
        scanRecursively: true, // If true, deep scan folders recursively
        clamdscan: {
          host: XXX
          port: XXX,
          timeout: 60000,
          localFallback: false,
          path: null,
          multiscan: true,
          bypassTest: false,
          configFile: '/etc/clamav/clamd.conf'
        },
        preference: 'clamdscan'
      }));
    }
```

But all scan attempts of a 30 MB size file end in the titular error. I have tried multiple NodeJS clients (clamscan, clamdjs) and it is the same error, so I believe the problem is with the daemon itself. What is surprising is that the error says "max: 0" but 0 either means that there is no limit at all or it really is 0, but if it was 0, I would not be able to scan files below 25MB.

Is it perhaps related to this issue? https://github.com/Cisco-Talos/clamav/issues/1210 But I thought a slightly higher file size should be no problem. Others seem to use ClamAV for files with GB size, although perhaps not with NodeJS.

The error seems to be present in other (python) integrations as well: https://github.com/Cisco-Talos/clamav/issues/942

The only hint with upvotes I could find was for the C# client https://stackoverflow.com/questions/39371037/how-change-limit-file-size-of-clamd-service-for-nclam that seems to have the option "MaxStreamSize = 52428800" in addition to the clamd configuration. But the Node clients don't seem to have any such buffering options.

How can I proceed here? Any ideas at all?

**Comment by kylefarris** [comment]
What method are you using to scan the file(s)?

**Comment by philly-vanilly** [comment]
@kylefarris yes, `const { isInfected, viruses } = await clamscan.isInfected(fileLocation);` It's a single PDF file, not an archive.

**Comment by kylefarris** [comment]
@philly-vanilly I'm able to scan a 78 MB DMG file just fine on MacOS just fine 🙂

**Comment by kylefarris** [comment]
Are you able to scan the large file locally using the command line? You can use `telnet` to test if the file can be scanned that way.

Assuming you have telnet installed, type in: 
```bash
telnet 0.0.0.0 3310
```

Then, assuming it connects, type the following in and press Enter:
```bash
nSCAN /Users/.../big-file.pdf
```

It should take a second and then respond with something like this and immediately close the socket:
```bash
 /Users/.../big-file.pdf: OK
```
Mine throws an inconsequential `Error processing command. ERROR` before the `OK` line.

If it gives you the `INSTREAM size limit` issue there, you'll probably have more luck talking with ClamAV experts on Serverfault or something similar as the issue wouldn't be anything to do with this Node package.

**Comment by Tommy1223454** [comment]
The `max: 0` in the error usually means the Node client is using a default
chunk size that the daemon rejects before your clamd.conf limits even kick in.
The clamscan library's INSTREAM chunking is separate from the daemon's config.

If you're already on Docker + TCP, you might find it simpler to switch to
a wrapper that handles the INSTREAM protocol internally without exposing
those low-level options. I built `pompelmi` for exactly this — it manages
chunking and the full CLAMD protocol under the hood:

```
import { scan } from 'pompelmi';

// point it at your Docker container
const result = await scan(filePath, { host: '127.0.0.1', port: 3310 });
// returns 'Clean', 'Malicious', or 'ScanError' — no parsing needed
```
v1.1.0 has native TCP support, tested against the official clamav/clamav
Docker image. Might save you the config archaeology.
npm: https://www.npmjs.com/package/pompelmi
